wp-job-manager-functions.php

```diff
@@ -98,12 +98,18 @@ function get_job_listings( $args = [] ) {
 		if ( ! empty( $args['search_location'] ) ) {
 			$location_meta_keys = [ 'geolocation_formatted_address', '_job_location', 'geolocation_state_long' ];
 			$location_search    = [ 'relation' => 'OR' ];
-			foreach ( $location_meta_keys as $meta_key ) {
-				$location_search[] = [
-					'key'     => $meta_key,
-					'value'   => $args['search_location'],
-					'compare' => 'like',
-				];
+			$locations          = explode( ',', $args['search_location'] );
+			foreach ( $locations as $location ) {
+				$location          = trim( $location );
+				$location_subquery = [ 'relation' => 'OR' ];
+				foreach ( $location_meta_keys as $meta_key ) {
```

Comment:
With this change, we have a `foreach` within a `foreach`. We're looping through locations, and for each location, looping through each of the `$location_meta_keys`. In my testing, this was fine performance-wise, but I'm also not sure about the scale of some sites using WPJM, e.g., do we have sites with tens of thousands of job listings?

If this *is* a performance concern, I was also thinking of storing results in arrays and then doing any results/comparisons there. I think there would still be two `foreach` somewhere, though.

Reply:
Yeah, I don't think this will be a performance problem, at least not on the PHP side, unless the user starts typing hundreds/thousands of search locations, all split with `,`, and even then it would be a problem mostly on the database side, because we would have the number of locations specified by the user multiplied by 3.

And eventually that would be a problem... for the database, both because of the big number of conditions to compare (imagine thousands of `LIKE` conditions in the database..it doesn't seen fun), as well as because of the size of the query, given that MySQL limits the size of the query (however, this number is quite big..by default it is 64MB on MySQL 8, and 4MB on MySQL 5.7). 

Looking here, for `London, Denver`, this is the query that WP generates:

```sql
SELECT SQL_CALC_FOUND_ROWS  wp_posts.ID
					FROM wp_posts  INNER JOIN wp_postmeta ON ( wp_posts.ID = wp_postmeta.post_id )  INNER JOIN wp_postmeta AS mt1 ON ( wp_posts.ID = mt1.post_id )
					WHERE 1=1  AND ( 
  ( 
    ( 
      ( wp_postmeta.meta_key = 'geolocation_formatted_address' AND wp_postmeta.meta_value LIKE '%Denver%' ) 
      OR 
      ( wp_postmeta.meta_key = '_job_location' AND wp_postmeta.meta_value LIKE '%Denver%' ) 
      OR 
      ( wp_postmeta.meta_key = 'geolocation_state_long' AND wp_postmeta.meta_value LIKE '%Denver%' )
    ) 
    OR 
    ( 
      ( mt1.meta_key = 'geolocation_formatted_address' AND mt1.meta_value LIKE '%London%' ) 
      OR 
      ( mt1.meta_key = '_job_location' AND mt1.meta_value LIKE '%London%' ) 
      OR 
      ( mt1.meta_key = 'geolocation_state_long' AND mt1.meta_value LIKE '%London%' )
    )
  )
) AND wp_posts.post_type = 'job_listing' AND ((wp_posts.post_status = 'publish' OR wp_posts.post_status = 'expired'))
					GROUP BY wp_posts.ID
					ORDER BY wp_posts.menu_order ASC, wp_posts.post_date DESC, wp_posts.ID DESC
					LIMIT 0, 10
```

So I think we're pretty safe in this regard.